dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/resourcetable/util/ColumnNameUtils.java

```diff
@@ -29,122 +29,33 @@
  */
 package org.hisp.dhis.resourcetable.util;
 
-import lombok.experimental.UtilityClass;
+import java.util.regex.Pattern;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
 import org.apache.commons.lang3.StringUtils;
+import org.hisp.dhis.commons.util.TextUtils;
 
-@UtilityClass
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class ColumnNameUtils {
-
-  private static final int MAX_COLUMN_NAME_LENGTH = 127;
+  private static final Pattern PAT_COL_NAME_CHARS =
+      Pattern.compile("[.a-zA-Z0-9_+\\-/?@#$%^&\\*\"\\s,:]");
```

Comment:
I think that `\\s` allows not only spaces but also newline (and tabs?). To only allow spaces it should be `"\ "\` (but not 100% sure here).